lib/java/com/google/android/material/materialswitch/MaterialSwitch.java

```diff
@@ -369,8 +404,23 @@ private void refreshThumbDrawable() {
 
     updateDrawableTints();
 
-    super.setThumbDrawable(
-        DrawableUtils.compositeTwoLayeredDrawable(thumbDrawable, thumbIconDrawable));
+    final Drawable thumbWithIconDrawable;
+    if (VERSION.SDK_INT >= VERSION_CODES.M) {
+      thumbWithIconDrawable = DrawableUtils.compositeDrawables(
+          thumbDrawable, thumbIconDrawable, thumbIconSize, thumbIconSize);
+    } else {
+      if (thumbIconDrawable != null
```

Comment:
The default value of thumbIconSize is 0. In other words, this condition will be triggered by most of the default settings, which is wrong. 

I think it is fine to just ignore the value without leaving logs.

Reply:
The condition will be triggered only for API < 23 and only for drawables that are not `thumbIconSize`x`thumbIconSize`. Logs are necessary here: user should be informed about why the custom icon size has not been applied.

As for size, default `thumbIconSize` for M3 theme - 16dp (`@dimen/mtrl_switch_thumb_icon_size`). The default value in `MaterialSwitch` constructor will be changed to intrinsic in the next rebase.

Reply:
Thanks for the explanation. I'm wondering if we can just restrict the attribute to API 23+ so we don't have to worry about this here. Let me check that with the team.